src/opts/mod.rs

```diff
@@ -144,6 +144,7 @@ pub struct SslOpts {
     root_cert_path: Option<Cow<'static, Path>>,
     skip_domain_validation: bool,
     accept_invalid_certs: bool,
+    domain_override: Option<Cow<'static, str>>,
```

Comment:
Minor naming nit: I'd suggest including `tls` in the name, and using the term `hostname` rather than `domain`, since AIUI "hostname" is the term of art used in TLS.

```suggestion
    tls_hostname_override: Option<Cow<'static, str>>,
```

Reply:
good point, will adjust the naming!